Source/Core/VideoCommon/Present.cpp

```diff
@@ -292,15 +293,22 @@ float Presenter::CalculateDrawAspectRatio(bool allow_stretch) const
     return (static_cast<float>(m_backbuffer_width) / static_cast<float>(m_backbuffer_height));
 
   auto& vi = Core::System::GetInstance().GetVideoInterface();
-  const float aspect_ratio = vi.GetAspectRatio();
+  const float source_aspect_ratio = vi.GetAspectRatio();
 
-  if (aspect_mode == AspectMode::AnalogWide ||
+  // This will scale up the source ~4:3 resolution to its equivalent ~16:9 resolution
+  if (aspect_mode == AspectMode::ForceWide ||
       (aspect_mode == AspectMode::Auto && g_widescreen->IsGameWidescreen()))
   {
-    return AspectToWidescreen(aspect_ratio);
+    return SourceAspectRatioToWidescreen(source_aspect_ratio);
+  }
+  // For the "custom" mode, we force the exact target aspect ratio, without
+  // acknowleding the difference between the source aspect ratio and 4:3.
+  else if (aspect_mode == AspectMode::Custom)
+  {
+    return g_ActiveConfig.GetCustomAspectRatio();
```

Comment:
I haven't tested this but I suspect just ignoring the VI-timing derived aspect ratio will negatively affect games that use a nonstandard timing -- what immediately comes to mind is those cinema-style widescreen scenes in Rouge Squadron. While I agree it's useful to be able to set the desired aspect ratio, I'm wondering if this shouldn't be multiplied onto the VI timings instead of completely overriding them.

Reply:
Yeah, exactly what I expected:

Using the 'Force 4:3' option: https://github.com/dolphin-emu/dolphin/assets/4522237/5cd95da0-137c-421f-972f-7914cd28efb7
Using the 'Custom' option and entering 4/3: https://github.com/dolphin-emu/dolphin/assets/4522237/248456f4-b876-4103-ab50-61af05d75c37

Despite both being '4:3', they are different, because the former respects the VI timings and the latter doesn't. This is especially awkward if the game switches VI timings between eg. cutscenes and gameplay or whatever -- as this game does.

I see multiple options here:
- Instead of treating the custom aspect ratio as an absolute, treat it as a modifier onto the VI aspect ratio. Calculate the difference to the VI-timing provided 4:3 and multiply onto that -- note that this is exactly what the current 16:9 option does. Probably the cleanest solution, but means that the resulting image is not actually exactly the entered aspect ratio in most cases.
- Just ignore this and stick with the absolute aspect ratio and hope the user knows what they're doing. Note that this will make the Custom option somewhat incompatible with games like Rogue Leader here.
- Or perhaps have both of these options? Might just confuse users though.

Reply:
Wouldn't the 'Stretch' setting in a 4:3 act like 'Custom' set to 4:3? That was the main intention here. It's just to allow specific "stretch" values without actually needing to resize the window to that specific desired aspect ratio (which sometimes is near impossible, e.g. I have a 32:9 screen and use 21:9 game mods).
I'd be fine with somebody else adding a 'Stretch (relative)' setting later on which acts as a relative aspect ratio modifier (ratio), I think it's beyond the scope of this PR and would like to thing about it after.
This is stable for now and works for the intent it has. It's a very specific use case and already helps users deal with specific scenarios IMO.

Reply:
> Wouldn't the 'Stretch' setting in a 4:3 act like 'Custom' set to 4:3? (Maybe not).

Assuming you mean 'in a 4:3 window', yes that would be the same thing. If this is purely supposed to be a replacement for Stretch without having to fiddle the window into the correct size, then yes, this is implemented correctly as far as I can tell. In that case I would like to make that more obvious in the UI in some way though.

Reply:
Maybe just rename the option to "Stretch (Custom)"?

Reply:
I got the code for the two "Custom - Absolute" and "Custom - Relative" modes ready, we can make a follow up PR

Reply:
We discussed this at length in Discord, but to immortalize it here: Yes, a follow-up sounds fine to me.